### DISCUSSION TITLE
Running code after register without useEffect

### DISCUSSION THREAD

**Discussion by nfuaufagf** [react-hook-form/react-hook-form#7163] [Q&A]
I am trying to use onChange validation. It seems to work fine when the mode in useForm() is set to onChange, but I want to run trigger() with the onChange props of the input instead. However, when doing this, the input value given by the custom validation argument in the register() function is outdated. This also applies to the value provided by watch() and getValues(). One way to run trigger() after register would be to use useEffect, but this is a bit tedious to do for me. Is there be an alternative way to achieve this?

**Comment by ritikbanger** [comment]
Please share a codesandbox to understand.
